cmd/edge-proxy/main.go

```diff
@@ -167,16 +167,17 @@ func main() {
 				fmt.Printf("Starting edge HTTP proxy (proxyAddr=%s, proxyURI=%s)\n", proxyAddr, proxyURI)
 
 				proxyForEdge := func(req *http.Request) (*url.URL, error) {
```

Comment:
Sorry I forgot to mention this during the last PR, it's been a while since I've thought about `edge-proxy` so I forgot. Have you considered using the [`http.ProxyFromEnvironment`](https://golang.org/pkg/net/http/#ProxyFromEnvironment) function? This is a more standardized approach for configuring communication through an HTTP proxy and makes use of the `HTTP_PROXY` and `HTTPS_PROXY` environment variables for configuration. A lot of other tools respect these environment variables. I'd recommend using this and removing the extra command line argument. This block could go away and I think even revert changes to function signatures made previously and just pass `http.ProxyFromEnvironment` in [here](https://github.com/armPelionEdge/edge-proxy/blob/883e6cb16db4a5b97302c54ba36ee5566ea8557b/http/transport.go#L61) directly:

>ProxyFromEnvironment returns the URL of the proxy to use for a given request, as indicated by the environment variables HTTP_PROXY, HTTPS_PROXY and NO_PROXY (or the lowercase versions thereof). HTTPS_PROXY takes precedence over HTTP_PROXY for https requests.
>
> The environment values may be either a complete URL or a "host[:port]", in which case the "http" scheme is assumed. An error is returned if the value is a different form.
>
> A nil URL and nil error are returned if no proxy is defined in the environment, or a proxy should not be used for the given request, as defined by NO_PROXY.
>
> As a special case, if req.URL.Host is "localhost" (with or without a port number), then a nil URL and nil error will be returned.

Reply:
@zhucebuliaolongchuan FYI

Reply:
I attempted this but was unable to get it to work.  My guess it was one f two issues.  Either my inexperience with go, or the restrictive snap environment.  I was never able to get HTTP_PROXY set in the snap environment in such a way that edge-proxy had visability to it.